### PR TITLE
Add Swap Hand Items and Jump (simple-)events

### PR DESCRIPTION
**Pull request by jaylawl**
Props to Nicofisi helping me understand how to make pull requests <3

Target Minecraft versions: any
Requirements: The jump event requires paper
Related issues: /

Description:
SwapHandItems event is used for when the player is using the dedicated key (default F) to swap their main and offhand items. It only fires while not viewing an inventory.

The PlayerJumpEvent event differs from spigot's PlayerMoveEvent and is less resource straining.

Related issues:
https://github.com/SkriptLang/Skript/issues/1445
https://github.com/SkriptLang/Skript/issues/444

### CODE REVIEW

**Comment by Nicofisi** [comment]
Have you tested that the stuff like `event-player` works? How about `event-location` in the jump event? 

Was about to ask about `event-item` in the swap event but what would it return, hmm..

Also, worth documenting whether `event-player's tool` used in the event returns the items *after* swapping or before that?

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 1]
```diff
@@ -434,5 +436,23 @@
...
+					.since("2.2-dev38");
+		}
+		if (Skript.classExists("org.bukkit.event.player.PlayerSwapHandItemsEvent")) {
+			Skript.registerEvent("Hand Item Swap", SimpleEvent.class, PlayerSwapHandItemsEvent.class, "swap[ping of] [(hand|held)] item[s]")
+					.description("Called whenever a player swaps the items in their",
+							"main- and offhand slots. Does not require any items to be held.")
+					.examples("on swap hand items:",
+							"	event-player's tool is a diamond sword",
+							"	cancel event")
+					.since("2.2-dev38");
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java]: use. INSERT VERSION

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 2]
```diff
@@ -434,5 +436,23 @@
...
+			Skript.registerEvent("Jump", SimpleEvent.class, PlayerJumpEvent.class, "[player] jump[ing]")
+					.description("Called whenever a player jumps",
+							"This event requires PaperSpigot")
+					.examples("on jump:",
+							"	event-player does not have permission \"jump\"",
+							"	cancel event")
+					.since("2.2-dev38");
+		}
+		if (Skript.classExists("org.bukkit.event.player.PlayerSwapHandItemsEvent")) {
+			Skript.registerEvent("Hand Item Swap", SimpleEvent.class, PlayerSwapHandItemsEvent.class, "swap[ping of] [(hand|held)] item[s]")
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java]: what about:
```
hotbar (switch|swap) [to %itemtype%]
```
so you can do
```
on hotbar switch to diamond sword:
```

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 3]
```diff
@@ -434,5 +436,23 @@
...
 					.since("2.2-dev37");
 		}
+		if (Skript.classExists("com.destroystokyo.paper.event.player.PlayerJumpEvent")) {
+			Skript.registerEvent("Jump", SimpleEvent.class, PlayerJumpEvent.class, "[player] jump[ing]")
+					.description("Called whenever a player jumps",
+							"This event requires PaperSpigot")
+					.examples("on jump:",
+							"	event-player does not have permission \"jump\"",
+							"	cancel event")
+					.since("2.2-dev38");
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java]: use INSERT VERSION

**Comment by jaylawl** [comment]
@Nicofisi yes, event-player works fine in both events and in the jump event, event-location works, too. I haven't bothered to check for event-item in the swap event, because just checking for player's tool and player's offhand item works fine without any hacky stuff. e.g.:
```
on swap items:
    event-player's tool = diamond sword
    event-player's offhand tool = dirt
    cancel event
```

**Comment by Pikachu920** [comment]
but what if you want to get the new/previous tool?

**Comment by jaylawl** [comment]
The new/previous tool can't be any other than what's in the main- and offhand

**Comment by Pikachu920** [comment]
ignore some of the reviews related to hotbars above, I've just realized this is swapping your offhand/main slots. I've never really actually played 1.9+ lol

**Comment by jaylawl** [comment]
Sorry about the commit name stuff btw. I wasn't trying to be a dick about it, i was just trying to get my first pull request to work and didn't realize those names would show here ^^

**Comment by jaylawl** [comment]
all the kids know `asd` stands for `advanced skript ... device?` x)

**Comment by Nicofisi** [comment]
image
*shh XDD*

---
also next time when you're resolving some issue with one of your commits please reference it within the commit title, like `Add player jump event (#444)` :heart: or probably even only referencing it in the PR description on GitHub should be enough

Anyway, linking for the reference: #444

**Comment by ShaneBeee** [comment]
#### Suggestion: 
Something in the description of the player jump event, mentioning it requires Paper? I have a small feeling a LOT of people are going to complain that it doesn't work on their spigot server. (I know I would be super confused it it wasn't working)

File: src/main/java/ch/njol/skript/events/SimpleEvents.java (hunk 2 above)
Comment (Snow-Pyon) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java]: Accept multiple item types, for `or` lists.

**Comment by Nicofisi** [comment]
If you did test everything then the PR looks good to me, the only thing I would still add though is the info that `tool of player` returns the item *after* the swap (right?) - it's probably intuitive but in my opinion it's really worth noting

**Comment by jaylawl** [comment]
@Nicofisi no, all returns are from before the event executes

**Comment by Nicofisi** [comment]
@jaylawl huh, then it's especially worth documenting that as you can see lmao

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 4]
```diff
@@ -447,8 +447,12 @@
 		}
 		if (Skript.classExists("org.bukkit.event.player.PlayerSwapHandItemsEvent")) {
 			Skript.registerEvent("Hand Item Swap", SimpleEvent.class, PlayerSwapHandItemsEvent.class, "swap[ping of] [(hand|held)] item[s]")
-					.description("Called whenever a player swaps the items in their",
-							"main- and offhand slots. Does not require any items to be held.")
+					.description("Called whenever a player swaps the items in their main- and offhand slots.",
+						     "Works also when one or both of the slots are empty.",
+						     "The event is called before the items are actually swapped,",
+						     "so when you use the player's tool or player's offtool expressions,",
+						     "they will return the values before the swap -",
```
[the remark this answers is not in the document]
Comment (jaylawl) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java]: it connects to the following sentence

[the remark this answers is not in the document]
Comment (Blueyescat) [reply src/main/java/ch/njol/skript/events/SimpleEvents.java]: Nope, it works like the example i gave ^^

**Comment by bensku** [comment]
Tested to work, merging.

**Pull request merged**